### PR TITLE
feat(Server): exclude client scripts for `node` targets

### LINKED ISSUE #1256: webpack Multiple Targets
### Code

```js
// webpack.config.js
module.exports = [
  {
    entry : {
      app: './src/app'
    },
    output: {
      path: path.resolve(__dirname, 'dist/')
      filename: 'assets/js/app.js',
      chunkFilename: `assets/js/${getAssetName('[hash]', '[name]')}.js`
    },
    devServer: {
      hot: true
    }
  },
  {
    entry : {
      app: './src/app.server'
    },
    output: {
      path: path.resolve(__dirname, 'dist/server')
      filename: 'app.js',
      chunkFilename: '[name].js',
      libraryTarget: 'commonjs2'
    },
    target: 'node'
  }
];
```
### Run
```shell
npx webpack-dev-server
```

### Expected Behavior
- ~~_The client app entry add `webpack/hot/only-dev-server` `webpack/hot/dev-server` etc._~~
- The client app entry add https://github.com/webpack/webpack-dev-server/blob/master/client-src/default/index.js file, from at line: https://github.com/webpack/webpack-dev-server/blob/master/lib/util/addDevServerEntrypoints.js#L23
- The server app entry no add.

### Actual Behavior
All added.

### For Bugs; How can we reproduce the behavior?
Example edit: https://github.com/webpack/webpack-dev-server/blob/master/lib/util/addDevServerEntrypoints.js#L22
```diff
    [].concat(webpackOptions).forEach((wpOpt) => {
+     if (wpOpt.target === 'node') return;
      if (typeof wpOpt.entry === 'object' && !Array.isArray(wpOpt.entry)) {
        Object.keys(wpOpt.entry).forEach((key) => {
          wpOpt.entry[key] = devClient.concat(wpOpt.entry[key]);
        });
      } else if (typeof wpOpt.entry === 'function') {
        wpOpt.entry = wpOpt.entry(devClient);
      } else {
        wpOpt.entry = devClient.concat(wpOpt.entry);
      }
    });
```

### For Features; What is the motivation and/or use-case for the feature?
server side rendering

### PR DESCRIPTION
**Pull request by EloB**
```bash
git clone git@github.com:EloB/ssr.git
cd ssr/node_modules/webpack-dev-server # <- This is using my branch
npm install # Install all dependencies
npm run prepublish # To build all client scripts
cd ../../
npm run dev
```

### Type 

- [x] Feature

### Issues

- Fixes #1256 
- Fixes #1338

### SemVer

- [x] Minor

### CODE REVIEW

**Comment by EloB** [comment]
I saw another PR about this: https://github.com/webpack/webpack-dev-server/pull/1338
One problem with that is that it sends out websocket on hashes that isn't `node` target.

File: lib/Server.js [hunk 1]
```diff
@@ -68,6 +68,7 @@ function Server(compiler, options, _log) {
   }
 
   const addCompilerHooks = (comp) => {
+    if (!['async-node', 'node'].includes(comp.options.target)) return;
```
Comment (EloB) [inline lib/Server.js]: This is to prevent websocket to not emit `node` target specific hashes.

**Comment by EloB** [comment]
@evilebottnawi Did you read the description? 😄

> Need help writing the tests.

**Comment by evilebottnawi** [comment]
@EloB oh sorry, i don't have time right now for this :disappointed: maybe someone other can help with this

**Comment by EloB** [comment]
@evilebottnawi No worry 😄👍 Lets hope someone has time!

**Comment by michael-ciniawsky** [comment]
@EloB See https://github.com/webpack/webpack-dev-server/pull/1338/files#diff-5e0bdc9fe7d56299ec06583d181360a5 for a test case and as a starting point :)

**Comment by EloB** [comment]
If I have some time over this weekend I will try to fix this. I really want that hot reloading on server side :)

**Comment by evilebottnawi** [comment]
@jojo-apollo why we need test websocket emission issue here?

**Comment by jojo-apollo** [comment]
This PR supposedly supercedes #1338, by attempting to solve some websocket hot reload issues which that PR doesn't address.

I don't have this websocket issue myself so I'm asking @EloB for clarification on a way to replicate it.

**Comment by evilebottnawi** [comment]
@jojo-apollo maybe we can test this manually

**Comment by jojo-apollo** [comment]
It's difficult to test if we don't know his exact use case, my `node` target currently uses custom plugins, so I have a feeling his change will break that.

I just want to confirm if it's an issue with way he wrote his webpack config (and he's configured or used it wrong).

https://webpack.js.org/concepts/targets/

**Comment by evilebottnawi** [comment]
@jojo-apollo let's wait 72 hours on answer and feel free to recreate PR

**Comment by EloB** [comment]
I managed to write a isomorphic webpack config without any changes with hot reload on both backend and frontend. When I have time I might be able to publish it.

**Pull request closed without merging**